pulsar-broker/src/main/java/org/apache/pulsar/broker/resourcegroup/ResourceGroup.java

```diff
@@ -294,13 +318,15 @@ protected BytesAndMessagesCount getGlobalUsageStats(ResourceGroupMonitoringClass
     protected BytesAndMessagesCount updateLocalQuota(ResourceGroupMonitoringClass monClass,
                                                      BytesAndMessagesCount newQuota) throws PulsarAdminException {
         this.checkMonitoringClass(monClass);
-        BytesAndMessagesCount oldBMCount = new BytesAndMessagesCount();
+        BytesAndMessagesCount oldBMCount;
 
         final PerMonitoringClassFields monEntity = this.monitoringClassFields[monClass.ordinal()];
         monEntity.localUsageStatsLock.lock();
         oldBMCount = monEntity.quotaForNextPeriod;
         try {
             monEntity.quotaForNextPeriod = newQuota;
+            this.resourceGroupPublishLimiter.publishMaxByteRate = newQuota.bytes;
+            this.resourceGroupPublishLimiter.publishMaxMessageRate = (int) newQuota.messages;
```

Comment:
is this cast dangerous ? 

can we add a method to  `resourceGroupPublishLimiter` in order to not access directly the fields (better encapsulation) ?

Reply:
Thanks; fixed in iteration f925d4d